libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationRouteProvider.java

```diff
@@ -0,0 +1,26 @@
+package com.mapbox.services.android.navigation.v5.navigation;
+
+import android.location.Location;
+
+import com.mapbox.services.android.navigation.v5.route.RouteFetcher;
+import com.mapbox.services.android.navigation.v5.routeprogress.RouteProgress;
+
+class NavigationRouteProvider {
```

Comment:
`NavigationRouteProvider` is called "provider" but doesn’t have any method that returns an object. That's confusing IMO.

Reply:
Also I'm wondering that maybe we should even remove this class completely. The functionality included is only "delegating" through the `RouteFetcher`. Couldn't `checkForFasterRoute` and `removeRouteEngineListener` methods being part of `RouteFetcher`? I think, creating a new class here is not worth it.
IMO, objects and methods should live/be where their context is.